trunk/SUAVE/Methods/Center_of_Gravity/compute_component_centers_of_gravity.py

```diff
@@ -15,81 +15,101 @@
-    furnishings        = vehicle.furnishings
-    passenger_weights  = vehicle.passenger_weights
-    air_conditioner    = vehicle.air_conditioner
-    fuel               = vehicle.fuel
-    apu                = vehicle.apu
-    hydraulics         = vehicle.hydraulics
-    optionals          = vehicle.optionals
+
     
+
+
     span_location_mac                          = compute_span_location_from_chord_length(wing, wing.chords.mean_aerodynamic)
     
     #assume that 80% of the chord difference is from leading edge sweep
-    mac_le_offset                              = .8*np.sin(wing.sweeps.leading_edge)*span_location_mac  
-    chord_length_h_tail_35_percent_semi_span   = compute_chord_length_from_span_location(h_tail,.35*h_tail.spans.projected*.5)
-    chord_length_v_tail_35_percent_semi_span   = compute_chord_length_from_span_location(v_tail,.35*v_tail.spans.projected*.5)
-    
     #x distance from leading edge of root chord to leading edge of aerodynamic center
-    h_tail_35_percent_semi_span_offset         =.8*np.sin(h_tail.sweeps.quarter_chord)*.35*.5*h_tail.spans.projected             
-    v_tail_35_percent_semi_span_offset         =.8*np.sin(v_tail.sweeps.quarter_chord)*.35*.5*v_tail.spans.projected
-    
+    mac_le_offset                              = .8*np.sin(wing.sweeps.leading_edge)*span_location_mac
     wing.mass_properties.center_of_gravity[0]   = .3*wing.chords.mean_aerodynamic + mac_le_offset
-    h_tail.mass_properties.center_of_gravity[0] = .3*chord_length_h_tail_35_percent_semi_span + \
-        h_tail_35_percent_semi_span_offset
-    v_tail.mass_properties.center_of_gravity[0] = .3*chord_length_v_tail_35_percent_semi_span + \
-        v_tail_35_percent_semi_span_offset
     
+    if vehicle.wings.has_key('horizontal_stabilizer'):
+        h_tail             = vehicle.wings['horizontal_stabilizer']
+        chord_length_h_tail_35_percent_semi_span   = compute_chord_length_from_span_location(h_tail,.35*h_tail.spans.projected*.5)
+        h_tail_35_percent_semi_span_offset         =.8*np.sin(h_tail.sweeps.quarter_chord)*.35*.5*h_tail.spans.projected   
+        h_tail.mass_properties.center_of_gravity[0] = .3*chord_length_h_tail_35_percent_semi_span + \
+        h_tail_35_percent_semi_span_offset
+    else: 
+        print "no horizontal stabilizer"
+        
+    if vehicle.wings.has_key('vertical_stabilizer'):
+        v_tail             = vehicle.wings['vertical_stabilizer']
+        chord_length_v_tail_35_percent_semi_span   = compute_chord_length_from_span_location(v_tail,.35*v_tail.spans.projected*.5)
+        v_tail_35_percent_semi_span_offset         =.8*np.sin(v_tail.sweeps.quarter_chord)*.35*.5*v_tail.spans.projected
+        v_tail.mass_properties.center_of_gravity[0] = .3*chord_length_v_tail_35_percent_semi_span + \
+            v_tail_35_percent_semi_span_offset
+    else: 
+        print "no vertical stabilizer"
+   
+    # need to correct calculations for propulsor center of gravity
```

Comment:
I think that we need to look at the scripts in the Center_of_Gravity folder before the release. I remember Michael doing a quick fix for me with the propulsors but I am not 100% sure it works